### DISCUSSION TITLE
Instance does not scale

### DISCUSSION THREAD

**Discussion by ahmadamirahmadi007** [apache/cloudstack#9989] [Q&A] [answered]
##### ISSUE TYPE
 * Bug Report

##### CLOUDSTACK VERSION
cloudstack version 4.19.1.1 on ubuntu
KVM

##### SUMMARY
I use the KVM hypervisor and I want to scale up my instance to a higher compute offering while it is in the powered-on state. Currently, I have the following three compute offerings:
x500-12-256-512         cpu=500MHZ     min-Core=1    max-Core=2   min-Ram=256  max-Ram=512
x1000-23-512-1024     cpu=1000MHZ    min-Core=2    max-Core=3   min-Ram=512  max-Ram=1024
x1500-13-1025-1500   cpu=1500MHZ    min-Core=1    max-Core=3   min-Ram=1025  max-Ram=1500

I create an instance with the first compute offering and then upgrade it to the second compute offering using the CMK . This process works correctly even though I'm using KVM and my instance is powered on. However, when I try to further upgrade the same instance to the third compute offering, I encounter an error code 530 and the message "Unhandled exception".

##### STEPS TO REPRODUCE
https://gist.github.com/ahmadamirahmadi007/140fd64feaab16d442ee0d36b67334ca

##### EXPECTED RESULTS
I want to upgrade to any higher offering service

##### ACTUAL RESULTS
CMK returned me:
"jobresult": {
    "errorcode": 530,
    "errortext": "Unhandled exception"
    
In the UI, after clicking on the scale option, it gives me the following error:
It's not possible to scale up Instances that utilize KVM hypervisor with a fixed compute offering.

**Comment by slavkap** [comment]
@ahmadamirahmadi007 there is a fix [PR 4894](https://github.com/apache/cloudstack/pull/9894). It'll be included in CloudStack 4.19.2.0 and 4.20. I cannot recall correctly but as an alternative, the scale VM API call has to work using the CLI
I don't have a setup with the issue but can you also check that the third offering is custom with "Dynamic scaling enabled"?

**Comment by ahmadamirahmadi007** [comment]
this is my third offering and Dynamic scaling is enabled 
list serviceofferings id=b54f8fe1-0a73-427f-b888-041cc8de4e8d
{
  "count": 1,
  "serviceoffering": [
    {
      "cpuspeed": 1500,
      "created": "2024-11-26T05:45:32+0000",
      "defaultuse": false,
      "diskofferingdisplaytext": "Large Disk, 100 GB",
      "diskofferingid": "6ae55ab1-30ee-475b-badd-e9ca758a84b5",
      "diskofferingname": "Large",
      "diskofferingstrictness": false,
      "displaytext": "x1500-13-1025-1500",
      "dynamicscalingenabled": true,
      "encryptroot": false,
      "hasannotations": false,
      "id": "b54f8fe1-0a73-427f-b888-041cc8de4e8d",
      "iscustomized": true,
      "issystem": false,
      "isvolatile": false,
      "limitcpuuse": false,
      "name": "x1500-13-1025-1500",
      "offerha": false,
      "provisioningtype": "thin",
      "rootdisksize": 100,
      "serviceofferingdetails": {
        "maxcpunumber": "3",
        "maxmemory": "1500",
        "mincpunumber": "1",
        "minmemory": "1025"
      },
      "state": "Active",
      "storagetype": "shared"
    }
  ]
}

**Comment by slavkap** [comment]
@ahmadamirahmadi007, have you enabled the debug level in the logs? If it's enabled can you look for this job ID `805ea32a` there has to be a stack trace as to why it is failing

**Comment by ahmadamirahmadi** [comment]
https://gist.github.com/ahmadamirahmadi/b3856d5d0483561595d7822e615e946e
I had several errors. And in lines 22 and 23 he told me that "enableDynamicallyScaleVm":"false" why? and i dont have any domain's maxMemory limitation in my CS @slavkap

**Comment by slavkap** [comment]
@ahmadamirahmadi007, the message "Attaching memory device with size '525312' would exceed domain's maxMemory config size '524288'" comes from libvirt not a CloudStack's domain. From what I found you cannot exceed the memory from the max memory with hot plugged.

**Reply by ahmadamirahmadi007** [reply, marked as answer]
tnx. i found When we used Dynamic Service Offering after the creation of vm, KVM set maxMemory as size as the Maximum threshold of its Dynamic Service Offering and this limitation stayed with vm. For this reason, promotion to a higher service offering is possible when we have a value of memory less than the maximum value of the first Service Offering. This method is also true for subsequent upgrades when the vm is Running.
